test/e2e/storage/pod_add_delete_add_fast.go

```diff
@@ -0,0 +1,179 @@
+
+	"k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+var _ = SIGDescribe("Fast pod add delete and then add back", func() {
+	var (
+		c                 clientset.Interface
+		ns                string
+		pvc               *v1.PersistentVolumeClaim
+		nodeName          string
+		isNodeLabeled     bool
+		nodeKeyValueLabel map[string]string
+		nodeLabelValue    string
+	)
+
+	f := framework.NewDefaultFramework("pv")
+
+	BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if len(nodeList.Items) != 0 {
+			nodeName = nodeList.Items[0].Name
+		} else {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		if !isNodeLabeled {
+			nodeLabelValue := "pod_vol_add_del_add_" + string(uuid.NewUUID())
+			nodeKeyValueLabel = make(map[string]string)
+			nodeKeyValueLabel["pod_vol_add_del_add_label"] = nodeLabelValue
+			framework.AddOrUpdateLabelOnNode(c, nodeName, "pod_vol_add_del_add_label", nodeLabelValue)
+			isNodeLabeled = true
+		}
+
+		// Make sure there is a default storageclass
+		defaultScName := getDefaultStorageClassName(c)
+		verifyDefaultStorageClass(c, defaultScName, true)
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+
+		pvc = newClaim(test, ns, "default")
+		updatedPVC, err := c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		pvc = updatedPVC
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc).ToNot(Equal(nil))
+	})
+
+	framework.AddCleanupAction(func() {
+		if len(nodeLabelValue) > 0 {
+			framework.RemoveLabelOffNode(c, nodeName, "pod_vol_add_del_add_label")
+		}
+	})
+
+	AfterEach(func() {
+		framework.Logf("AfterEach: Cleaning up resources for fast add delete")
+
+		if c != nil {
+			if errs := framework.PVPVCCleanup(c, ns, nil, pvc); len(errs) > 0 {
+				framework.Failf("AfterEach: Failed to delete PVC and/or PV. Errors: %v", utilerrors.NewAggregate(errs))
+			}
+			pvc, nodeName, isNodeLabeled, nodeLabelValue = nil, "", false, ""
+			nodeKeyValueLabel = make(map[string]string)
+		}
+
+	})
+
+	It("should test that quickly adding, deleting and then adding back a pod does not stop pod creation", func() {
+		podCommand := "while true; do date; date >>/mnt/volume1; sleep 1; done"
+		if pvc == nil {
+			framework.Failf("PVC should not be nil for the test")
+		}
+		pod1 := createPodWithNodeAffinity(pvc.Name, nodeKeyValueLabel, podCommand)
+		pod1, err := c.CoreV1().Pods(ns).Create(pod1)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		time.Sleep(10)
+
+		// now lets delete the pod
+		podDeleteErr := c.CoreV1().Pods(ns).Delete(pod1.Name, nil)
```

Comment:
`framework.DeletePodWithWait` waits for the pod to be deleted.

Reply:
Again, I don't want to wait for pod to be deleted before creating new pod.  The reason is, we want to do add, delete and add back quickly without waiting for each action to be complete, except last one.

Reply:
ack